docs/notebooks/frequency_analysis.ipynb

```diff
@@ -0,0 +1,227 @@
+    "\n",
+    "To facilitate this type of analysis on a large number of time series from model simulations or observations, `xclim` packs a few common utility functions. In the following example, we're estimating the 95th percentile of the daily precipitation maximum over the May-October period using a Generalized Extreme Value distribution.\n",
+    "\n",
+    "Note that at the moment, all frequency analysis functions are hard-coded to operate along the `time` dimension.\n",
+    "\n",
+    "Let's first create a synthetic time series of daily precipitation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import xarray as xr\n",
+    "from xclim.indices.stats import frequency_analysis, fit, fa, parametric_quantile\n",
+    "from xclim.indices.generic import select_resample_op\n",
+    "from xclim.core.missing import missing_pct\n",
+    "from scipy.stats import gamma, bernoulli\n",
+    "\n",
+    "# Create synthetic daily precipitation time series (mm/d)\n",
+    "n = 50 * 366\n",
+    "start = np.datetime64(\"1950-01-01\")\n",
+    "time = start + np.timedelta64(1, \"D\") *  range(n)\n",
+    "#time = xr.cftime_range(start=\"1950-01-01\", periods=n)\n",
+    "\n",
+    "# Generate wet (1) /dry (0) days, then multiply by rain magnitude.\n",
+    "wet = bernoulli.rvs(.1, size=n)\n",
+    "intensity = gamma(a=4, loc=1, scale=6).rvs(n)\n",
+    "pr = xr.DataArray(wet * intensity, dims=(\"time\",), coords={\"time\": time}, attrs={\"units\": \"mm/d\"})\n",
+    "pr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "The `frequency_analysis` function combines all the necessary steps to estimate our design value:\n",
+    "1. Extract May to October period (`month=[5,6,7,8,9,10]`)\n",
+    "2. Extract maxima (`mode=\"max\"`)\n",
+    "3. Fit the GEV distribution on the maxima (`dist=\"genextreme\"`)\n",
+    "4. Compute the value exceeded, on average, once every 20 years (`t=20`)\n",
+    "\n",
+    "Note that `xclim` essentially wraps `scipy.stats` distributions, so many distributions like `norm`, `gumbel_r`, `lognorm`, etc. are supported."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Compute the design value\n",
+    "frequency_analysis(pr, t=20, dist=\"genextreme\", mode=\"max\", freq=\"Y\", month=[5,6,7,8,9,10] )"
```

Comment:
Is there a reason why you use 'Y' and not 'YS' like we usually do?

Reply:
None, in this case it doesn't really matter.